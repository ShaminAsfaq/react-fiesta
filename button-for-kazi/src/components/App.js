import React from 'react';
import Button from './Button';
import LoggedIn from './LoggedIn';
import LoggedOut from './LoggedOut';
import Duration from './Duration';

class App extends React.Component {

    timeStyleLeft = {
        paddingLeft: "30px",
        paddingTop: "30px"
    };

    timeStyleRight = {
        paddingTop: "40px"
    }

    durationStyleMiddle = {
        paddingTop: "30px",
        paddingLeft: "30px",
    }

    render() {
        return (
            <div>
                <div className="ui secondary pointing menu">
                    <Button />
                </div>
                <div className="ui statistics">
                    <div style={this.timeStyleLeft}>
                        <LoggedIn/>
                    </div>
                    <div className="ui statistics">
                        <div className="statistic" style={this.durationStyleMiddle}>
                            <Duration/>
                        </div>
                    </div>
                    <div style={this.timeStyleLeft}>
                        <LoggedOut/>
                    </div>
                </div>
            </div>
        );
    }
}

export default App;

