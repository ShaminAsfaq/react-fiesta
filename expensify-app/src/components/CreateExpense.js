import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

class CreateExpense extends React.Component {
    onSubmit = (expense) => {
        // props.dispatch(addExpense(expense))
        this.props.createExpense(expense)
        this.props.history.push('/')
    };

    render() {
        return (
            <div>
                <h1>Create Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        createExpense: (expense) => {
            return (
                dispatch(addExpense(expense))
            );
        }
    }
}

var ConnectedCreateExpense = connect(undefined, mapDispatchToProps)(CreateExpense)

export { 
    ConnectedCreateExpense as default,
    CreateExpense
}



