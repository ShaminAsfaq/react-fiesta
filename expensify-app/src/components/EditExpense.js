import React from 'react';

const EditExpense = (props) => {   
    console.log(props);
    return (
        <div>
            Current ID is {props.match.params.id}
        </div>
    );
}

export {
    EditExpense as default
}
