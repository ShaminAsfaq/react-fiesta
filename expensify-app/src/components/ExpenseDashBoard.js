import React from 'react';

const ExpenseDashBoard = () => {
    return (
        <div>
            Component
        </div>
    );
}

export {
    ExpenseDashBoard as default
}
