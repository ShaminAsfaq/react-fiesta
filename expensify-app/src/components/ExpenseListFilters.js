import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target);
    };
    onSortChange = (e) => {
        if (e.target.value==='date') {
            this.props.sortByDate();
        } else if (e.target.value==='amount') {
            this.props.sortByAmount();
        }
    };

    render() {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange}/>

                <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>

                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={({ startDate, endDate }) => {
                        this.props.setStartDate(startDate);
                        this.props.setEndDate(endDate);
                    }}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={(calendarFocused) => {
                        this.setState(() => {
                            return {
                                calendarFocused
                            }
                        });
                    }}
                    isOutsideRange={() => false}
                    numberOfMonths={1}
                    showClearDates={true}
                />

            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => {
            return (
                dispatch(setTextFilter(text))
            )
        },
        sortByDate: () => {
            return (
                dispatch(sortByDate())
            )
        },
        sortByAmount: () => {
            return (
                dispatch(sortByAmount())
            )
        },
        setStartDate: (startDate) => {
            return (
                dispatch(setStartDate(startDate))
            )
        },
        setEndDate: (endDate) => {
            return (
                dispatch(setEndDate(endDate))
            )
        }
    }
}

const ConnectedExpenseListFilters = connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);

export {
    ConnectedExpenseListFilters as default,
    ExpenseListFilters
}

