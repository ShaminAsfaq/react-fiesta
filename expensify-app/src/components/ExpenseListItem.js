import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div>
            <h3>{description}</h3>
            <p>{amount}-{createdAt}</p>
            <Link to={`/edit/${id}`}>
                <button>
                    Edit
                </button>
            </Link>
        </div>
    );
}

export {
    ExpenseListItem as default
}

