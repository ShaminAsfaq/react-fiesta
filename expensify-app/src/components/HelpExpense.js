import React from 'react';

const HelpExpense = () => {
    return (
        <div>
            Get Help!!
        </div>
    );
}

export {
    HelpExpense as default
}

