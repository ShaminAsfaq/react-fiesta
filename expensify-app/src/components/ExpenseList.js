import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => {
    return (
        <div>
            {
                props.expenses.length === 0 ? (
                    <p>Nothing to render</p>
                ) : (
                    props.expenses.map((item) => {
                        return (
                            <ExpenseListItem key={item.id} {...item}/>
                        );
                    })
                )
            }
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
};

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList);

export {
    ConnectedExpenseList as default,
    ExpenseList
}

