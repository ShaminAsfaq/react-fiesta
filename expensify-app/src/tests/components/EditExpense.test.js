import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { EditExpense } from '../../components/EditExpense';

let editExpenseSpy, removeExpenseSpy, history, wrapper;

beforeEach(() => {
    editExpenseSpy = jest.fn();
    removeExpenseSpy = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpense 
                            editExpense={editExpenseSpy}
                            removeExpense={removeExpenseSpy}
                            history={history}
                            expense={expenses[2]}
                        />
                    );
})

test('Testing EditExpense Render', () => {
    expect(wrapper).toMatchSnapshot();
})

test('Testing EditExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpenseSpy).toHaveBeenCalledWith(expenses[2].id, expenses[2]);
})

test('', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpenseSpy).toHaveBeenCalledWith({
        id: expenses[2].id
    });
})