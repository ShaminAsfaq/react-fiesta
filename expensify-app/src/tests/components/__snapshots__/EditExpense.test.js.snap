// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditExpense Render 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 75,
        "createdAt": 345600000,
        "description": "Fame",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
